pandas/tests/extension/test_arrow.py

```diff
@@ -3239,13 +3239,71 @@ def test_arrow_floordiv():
 
 
 def test_arrow_floordiv_large_values():
-    # GH 55561
+    # GH 56645
     a = pd.Series([1425801600000000000], dtype="int64[pyarrow]")
     expected = pd.Series([1425801600000], dtype="int64[pyarrow]")
     result = a // 1_000_000
     tm.assert_series_equal(result, expected)
 
 
+def test_arrow_floordiv_large_integral_result():
+    # GH 56676
+    a = pd.Series([18014398509481983, -9223372036854775808], dtype="int64[pyarrow]")
+    result = a // 1
+    tm.assert_series_equal(result, a)
+
+
+def test_arrow_floordiv_larger_divisor():
+    # GH 56676
+    a = pd.Series([-23], dtype="int64[pyarrow]")
+    result = a // 24
+    expected = pd.Series([-1], dtype="int64[pyarrow]")
+    tm.assert_series_equal(result, expected)
+
+
+def test_arrow_floordiv_integral_invalid():
+    # GH 56676
+    a = pd.Series([-9223372036854775808], dtype="int64[pyarrow]")
```

Comment:
Can we parametrize the pyarrow integral types here as well? Not sure if pyarrow provides a negative limit for each type but would be ideal to leverage that if available

Reply:
Good idea, will add this.